magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/SNSDiscoveryIT.java

```diff
@@ -0,0 +1,72 @@
+package io.openraven.magpie.plugins.aws.discovery.services;
+
+
+import io.openraven.magpie.api.Emitter;
+import io.openraven.magpie.api.MagpieEnvelope;
+import io.openraven.magpie.plugins.aws.discovery.services.base.BaseAWSServiceIT;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.times;
+
+@ExtendWith(MockitoExtension.class)
+public class SNSDiscoveryIT extends BaseAWSServiceIT {
+  private static final String CF_SNS_TEMPLATE = "/template/sns-subscription-template.yml";
+  private final SNSDiscovery snsDiscovery = new SNSDiscovery();
+
+  @Mock
+  private Emitter emitter;
+
+  @Captor
+  private ArgumentCaptor<MagpieEnvelope> envelopeCapture;
+
+  @BeforeAll
+  public static void setup() {
+    updateStackWithResources(CF_SNS_TEMPLATE);
+  }
+
+  @Test
+  public void testSNSDiscovery() {
+    snsDiscovery.discover(
+      MAPPER,
+      SESSION,
+      BASE_REGION,
+      emitter,
+      LOGGER,
+      ACCOUNT
+    );
+
+    // Verify
+    Mockito.verify(emitter, times(2)).emit(envelopeCapture.capture());
+    var invocations = envelopeCapture.getAllValues();
+    assertEquals(2, invocations.size());
+    assertTopic(invocations.get(0));
+    assertSubscription(invocations.get(1));
```

Comment:
Please consider splitting this to 2 test cases

Reply:
This will be possible only if we will expose methods `discoverTopics` and `discoverSubscriptions`.
Not sure if we want it for test purpose. Initially I was designing cumulative approach to test input/output for available exposed method.
This decrease execution time and leading to the comparable coverage.
But it could fail because issue in either topic or subscription - at this point of IT testing that should be acceptable.
I will propose separate refactoring if required later.